### PR TITLE
[feat] Pass parent CI image name and ReFrame verbosity level in the generated Gitlab child pipelines

### LINKED ISSUE #2134: Emit image name in generated CI file
Hello,

As per the tutorial of integrating ReFrame into CI, ReFrame dynamically generates this `pipeline.yml` file for the tests user wants to do. Typically, this file has following contents:

```
cache:
  key: ${CI_COMMIT_REF_SLUG}
  paths:
  - rfm-stage/${CI_COMMIT_SHORT_SHA}
stages:
- rfm-stage-0
- rfm-stage-1
Test-1:
  stage: rfm-stage-0
  script:
  <snippet>
  artifacts:
    paths:
    - Test-1-report.json
  needs: []
Test-2:
  stage: rfm-stage-1
  script:
  <snippet>
  artifacts:
    paths:
    - Test-2-report.json
  needs:
  - Test-1
```

The problem is that the generated CI file does not emit `image` name on the top of the file. So, docker executor fallback to default image defined in the runner configuration for these tests. Consequently, it works only when the CI pipeline uses default runner image and results in failed CI jobs when different image is used.

Usually, as developers, we use customised images for CI jobs which include all the packages, files and dependencies. Hence, it would be useful if ReFrame can emit `image` directive on the top of the generated CI file. GitLab gives the `CI_JOB_IMAGE` environment variables within CI and it can be used to emit the requested directive.

### PR DESCRIPTION
**Pull request by mahendrapaipuri**
- This PR emits docker image name, if defined in the parent CI script, in the generated CI scripts. 
- If the user chose to increase the verbosity in the parent CI script, the child scripts will also inherit the same level of verbosity. In the current implementation, there is no verbosity and it is hard for the user to debug the test failures. 
- Tested on simple [CI pipeline](https://gitlab.com/ska-telescope/sdp/ska-sdp-benchmark-tests/-/jobs/1513187407#L216).

Fixes #2134

### CODE REVIEW

**Comment by teojgo** [comment]
Ok to test

**Comment by mahendrapaipuri** [comment]
Does it sound logical to add a new configuration section for CI pipelines to define parameters like `before-script`, `after-script`, `expire_in` (for artifacts) so that they will be emitted in the generated `yaml` files?

**Review by vkarak — CHANGES_REQUESTED** [review]
@mahendrapaipuri thanks a lot for your PR! I have just a couple of minor comments. Also we should probably document these additions somewhere in the user documentation.

File: reframe/frontend/ci.py [hunk 1]
```diff
@@ -51,6 +54,16 @@ def rfm_command(testcase):
         },
         'stages': []
     }
+
+    # Name of the image used for CI. If user does not explicitly provide
+    # image keyword on the top of CI script, this variable does not exist
+    image_name = os.getenv('CI_JOB_IMAGE')
+    if image_name:
+        json = {
+            'image': image_name,
```
[the remark this answers is not in the document]
Comment (mahendrapaipuri) [reply reframe/frontend/ci.py]: Well, I did a test and it looks like placement of `image` keyword is indeed not critical. Changed to your suggestion.

File: reframe/frontend/ci.py [hunk 2]
```diff
@@ -18,6 +19,7 @@ def _emit_gitlab_pipeline(testcases):
     prefix = 'rfm-stage/${CI_COMMIT_SHORT_SHA}'
     checkpath = config.get('general/0/check_search_path')
     recurse = config.get('general/0/check_search_recursive')
+    verbosity = 'v' * int(config.get('general/0/verbose'))
```
[the remark this answers is not in the document]
Comment (mahendrapaipuri) [reply reframe/frontend/ci.py]: Cheers. Changed it too.

**Review by vkarak — APPROVED** [review]

**Pull request merged**